tasks.py

```diff
@@ -0,0 +1,220 @@
+        '--target "{}"'.format(output),
+        '--no-compile',
+        '--no-deps'
+    ]
+    c.run(" ".join(args), echo=True)
+    os.unlink(tmp.name)
+
+    print('--> Copying source files')
+    copy_tree("src", output)
+
+    if ziparchive is not None:
+        print('--> Compressing to {}'.format(ziparchive))
+        zip_folder_to_file(output, ziparchive)
+
+def _read_url(response :HTTPResponse) -> str:
+    charset = response.headers.get_content_charset('utf-8')
+    raw_data = response.read()
+    return raw_data.decode(charset)
+
+
+def _get_filename_from_url(url: str) -> str:
+    return url.split("/")[-1]
+
+
+def _pull_protobufs_internal(c, selection: str, silent: bool = False):
+    target_dir = os.path.join(BASE_DIR, "protobuf_files", "proto")
+    list_file = os.path.join(BASE_DIR, "protobuf_files", f"protobuf_{selection}.txt")
+
+    try:
+        rmtree(target_dir)
+    except Exception:
+        pass  # directory probably just already exists
+
+    os.makedirs(target_dir, exist_ok=True)
+
+    with open(list_file, "r") as file:
+        urls = filter(None, file.read().split("\n"))  # filter(None, ...) is used to strip empty lines from the collection
+
+    for url in urls:
+        if not silent:
+            print("Retrieving: " + url)
+
+        file_name = _get_filename_from_url(url)
+
+        response = urlopen(url)
+        data = _read_url(response)
+
+        # needed to avoid packages of the form ...steam_auth.steamclient_pb2
+        if ".steamclient.proto" in file_name:
+            file_name = file_name.replace(".steamclient.proto", ".proto")
+        if ".steamclient.proto" in data:
+            data = data.replace(".steamclient.proto", ".proto")
+
+        if "cc_generic_services" in data:
+            data = data.replace("cc_generic_services", "py_generic_services")
+
+        if selection == "webui":
+            # lil' hack to avoid name collisions; the definitions are (almost) identical so this shouldn't break anything
+            data = data.replace("common_base.proto", "steammessages_unified_base.proto")
+            data = data.replace("common.proto", "steammessages_base.proto")
+
+        # force proto2 syntax if not yet enforced
+        if "proto2" not in data:
+            data = f'syntax = "proto2";\n' + data
+
+        with open(os.path.join(target_dir, file_name), "w") as dest:
+            dest.write(data)
+
+
+@task
+def InstallProtoc(c):
+    if os.path.exists(PROTOC_DIR) and os.path.isdir(PROTOC_DIR):
+        print("protoc directory already exists, remove it if you want to reinstall protoc")
+        return
+
+    os.makedirs(PROTOC_DIR)
+
+    resp = requests.get(PROTOC_DOWNLOAD_URL, stream=True)
+    resp.raise_for_status()
+
+    with zipfile.PyZipFile(BytesIO(resp.content)) as zipf:
+        zipf.extractall(PROTOC_DIR)
+
+    print("protoc successfully installed")
+
+
+#for whatever reason if i give this an _ in the name it can't find it. i have no idea why. so TitleCase
```

Comment:
Huh. Well, that's good to know. I actually prefer TitleCase to snake_case and imo it switching from '_' to '-' is not a great idea on `invoke`'s part because it's not clear what's going on (as that comment clearly demonstrates), but now i know how to use the optional fields, so that's nice.